code/modules/mob/living/simple_mob/subtypes/animal/space/bluespaceworm.dm

```diff
@@ -0,0 +1,432 @@
+	set desc = "Extends your teeth for 30 seconds so that you can chew through mobs and structures alike."
+	set category = "Abilities"
+
+	if(world.time < time_maw_opened + maw_cooldown)
+		if(open_maw)
+			to_chat(src, "<span class='notice'>You retract your teeth.</span>")
+			time_maw_opened -= maw_cooldown / 2	// Recovers half cooldown if you end it early manually.
+		else
+			to_chat(src, "<span class='notice'>You are too tired to do this..</span>")
+		set_maw(FALSE)
+	else
+		set_maw(!open_maw)
+
+/mob/living/simple_mob/animal/space/bluespace_worm/proc/set_maw(var/state = FALSE)
+	open_maw = state
+	if(open_maw)
+		time_maw_opened = world.time
+		movement_cooldown = initial(movement_cooldown) + 1.5
+	else
+		movement_cooldown = initial(movement_cooldown)
+	update_icon()
+
+/mob/living/simple_mob/animal/space/bluespace_worm/death()
+	..()
+
+	DumpStomach()
+
+	if(previous)
+		previous.death()
+
+/mob/living/simple_mob/animal/space/bluespace_worm/handle_special()	// Processed in life. Nicer to have it modular incase something in Life change(d)(s)
+	..()
+
+	if(world.time > time_maw_opened + maw_cooldown)	// Auto-stop eating.
+		if(open_maw)
+			to_chat(src, "<span class='notice'>Your jaws cannot remain open..</span>")
+			set_maw(FALSE)
+
+	if(next && !(next in view(src,1)) && !z_transitioning)
+		Detach(1)
+
+	if(stat == DEAD && sever_chunks) // Dead chunks fall off and die immediately if we sever_chunks
+		if(previous)
+			previous.Detach(1)
+		if(next)
+			Detach(1)
+
+	if(prob(stomachProcessProbability))
+		ProcessStomach()
+
+	update_icon()
+
+	return
+
+/mob/living/simple_mob/animal/space/bluespace_worm/CanPass(atom/movable/mover, turf/target)
+	if(istype(mover, /mob/living/simple_mob/animal/space/bluespace_worm/head))
+		var/mob/living/simple_mob/animal/space/bluespace_worm/head/H = mover
+		if(H.previous == src)
+			return FALSE
+
+	if(istype(mover, /mob/living/simple_mob/animal/space/bluespace_worm))	// Worms don't run over worms. That's weird. And also really annoying.
+		return TRUE
+	else if(istype(mover, /mob/living/simple_mob/animal/space/space_worm))	// Including other worm type.
+		return TRUE
+	else if(src.stat == DEAD && !istype(mover, /obj/item/projectile))	// Projectiles need to do their normal checks.
+		return TRUE
+	return ..()
+
+/mob/living/simple_mob/animal/space/bluespace_worm/Destroy() // If a chunk is destroyed, kill the back half.
+	DumpStomach()
+	if(previous)
+		previous.Detach(1)
+	if(next)
+		next.previous = null
+		next = null
+	..()
+
+/mob/living/simple_mob/animal/space/bluespace_worm/Moved(atom/old_loc, direction, forced = FALSE)
+	. = ..()
+	if(previous)
+		if(previous.z != z)
+			previous.z_transitioning = TRUE
+		else
+			previous.z_transitioning = FALSE
+		previous.forceMove(old_loc)	// None of this 'ripped in half by an airlock' business.
+	update_icon()
+
+/mob/living/simple_mob/animal/space/bluespace_worm/head/Bump(atom/obstacle)
+	if(open_maw && !stat && obstacle != previous)
+		spawn(1)
```

Comment:
what's this spawn() for even ? spawn() should rarely be used because mostly you don't need to in 99% of cases

Reply:
In this case, using `spawn()` creates a delay between the worm Bumping something/one and it actually consuming them. This could lead to circumstances where the bumped thing moves away from the worm, breaking any adjacency requirements. That is to say, the spawn here is bad.